File: app/src/main/java/com/example/myapplication/testdata/SecondActivity.java
package com.example.myapplication.testdata;

import android.app.Activity;
import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.TextView;

import com.example.myapplication.R;

/**
 * create by xujixiao on 2019-09-10 14:03
 */
public class SecondActivity extends Activity {


    public static void start(Context context) {
        Intent starter = new Intent(context, SecondActivity.class);
//        starter.putExtra();
        context.startActivity(starter);
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_first_layout);
        TextView text = findViewById(R.id.tv_test_first);
        text.setText("第二个");
        text.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent();
                intent.putExtra("data", "second_data-20");
                setResult(20, intent);
                finish();
            }
        });

    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        Log.d("xujixiao", "Second---requestCode---" + requestCode + "---resultCode--" + resultCode + "----data---" + data.getStringExtra("data"));
    }
}

File: app/src/main/java/com/example/myapplication/kotin/SecondActivity.kt
package com.example.myapplication.kotin

import android.app.Activity
import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.util.Log
import com.example.myapplication.R
import kotlinx.android.synthetic.main.activity_main.*
import kotlinx.android.synthetic.main.include_test_layout.*
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.GlobalScope
import kotlinx.coroutines.launch
import kotlinx.coroutines.withContext

/**
 * create by xujixiao on 2019-05-22 15:45
 */
class SecondActivity : Activity() {

    companion object {
        fun start(context: Context) {
            val starter = Intent(context, SecondActivity::class.java)
//            starter.putExtra();
            context.startActivity(starter)
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        tv_test_kotlin.text = "测试kotlin的扩展用法"
//        myName.text="我的名字"
        myName.text="我的名字"
        tv_test_kotlin.setOnClickListener {
            //            主线程使用协程实例
            Log.d("xujixiao", "开始启动协程")
//            GlobalScope.launch(Dispatchers.Main) {
//                val string = sleepTime()
//                Toast.makeText(this@SecondActivity, string, Toast.LENGTH_LONG).show()
//                tv_test_kotlin.text = string
//            }

            GlobalScope.launch(Dispatchers.IO) {
                val text = sleepTime()
//                tv_test_kotlin.text=text
                withContext(Dispatchers.Main) {
                    tv_test_kotlin.text = text
                }
            }
        }
    }

    suspend fun sleepTime(): String {
        Thread.sleep(4000)
        return "睡醒了"
    }
}